src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -150,21 +150,18 @@ private JsonArray listTemplates(String pageToken, JsonArray jsonElements) {
     AribaResponseContainer responseContainer = null;
     try {
       responseContainer = aribaServices.fetchAribaResponse(viewTemplatesURL, accessToken);
-    } catch (AribaException | IOException | InterruptedException e) {
-      LOG.error("Unable to fetch response", e);
+    } catch (AribaException | InterruptedException e) {
+      throw new IOException("Unable to fetch response", e);
     }
     InputStream responseStream = responseContainer.getResponseBody();
     BufferedReader reader = new BufferedReader(new InputStreamReader(responseStream, StandardCharsets.UTF_8));
     String result = reader.lines().collect(Collectors.joining(""));
-    JsonObject jsonObject = gson.fromJson(result, JsonObject.class);
+    JsonObject jsonObject = GSON.fromJson(result, JsonObject.class);
     JsonElement nextPageToken = jsonObject.get(TOKEN);
-    try {
-      if (!nextPageToken.isJsonNull()) {
+      if (null != jsonObject.get(TOKEN)) {
```

Comment:
nit: don't need to do a get again.
if (nextPageToken != null) {
...

Reply:
done